File: 백준/Silver/11057. 오르막 수/오르막 수.java
    import java.util.*;
    import java.io.*;
    public class Main {

    	public static void main(String[] args) throws IOException {
    	BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
    	BufferedWriter bw=new BufferedWriter(new OutputStreamWriter(System.out));
    	int num=Integer.parseInt(br.readLine());
    	long arr[][]=new long[num][10];
    	for(int i=0;i<10;i++) {
    		arr[0][i]=1;
     	}
    	for(int j=1;j<num;j++) {
    		for(int k=0;k<10;k++) {
    			for(int l=0;l<=k;l++) {
    				arr[j][k]+=arr[j-1][l];
    				arr[j][k]%=10007;
    			}
    		}
    	}
    	long sum=0;
    	for(int q=0;q<10;q++) {
    		sum+=arr[num-1][q];
    		sum%=10007;
    	}
    	System.out.print(sum);
}
    	}
    

File: 백준/Silver/11057. 오르막 수/오르막 수.py
n = int(input())
arr = [[0]*10 for _ in range(n)]
for i in range(10):
    arr[0][i] = 1
    
for i in range(1,n):
    for j in range(10):
        for k in range(j+1):
            arr[i][j] += arr[i-1][k]
            arr[i][j] %= 10007

s = sum(arr[n-1])
print(s%10007)